packages/@aws-cdk/aws-ecs-patterns/lib/base/application-load-balanced-service-base.ts

```diff
@@ -199,14 +215,33 @@ export abstract class ApplicationLoadBalancedServiceBase extends cdk.Construct {
       port: 80
     };
 
+    if (props.protocol !== undefined && props.protocol !== ApplicationProtocol.HTTPS && props.certificate !== undefined) {
+      throw new Error('The HTTPS protocol must be used when a certificate is given');
+    }
+    const protocol = props.certificate ? ApplicationProtocol.HTTPS : (props.protocol ? props.protocol : ApplicationProtocol.HTTP);
+
     this.listener = this.loadBalancer.addListener('PublicListener', {
-      port: props.certificate !== undefined ? 443 : 80,
+      protocol,
       open: true
     });
     this.targetGroup = this.listener.addTargets('ECS', targetProps);
 
-    if (props.certificate !== undefined) {
-      this.listener.addCertificateArns('Arns', [props.certificate.certificateArn]);
+    if (protocol === ApplicationProtocol.HTTPS) {
+      if (typeof props.domainName === 'undefined' || typeof props.domainZone === 'undefined') {
```

Comment:
Should the domain name and zone not only be required if `certificate` is not given?

Reply:
Yeah I went back and forth on this.  Thoughts @pkandasamy91?

Pros: It prevents a misconfiguration where the HTTPS endpoint ends up not addressable by a custom domain name.  With ALB, an HTTPS endpoint always has to be addressed by a custom domain name for it to be usable.  Accessing the ALB over its default xxx.elb.amazonaws.com DNS name would result in an SSL cert mismatch, since ALB won't assign a cert for xxx.elb.amazonaws.com.  When a cert is given by the user, I can't validate that the given cert's domain name matches the given domain name, but I can at least ensure there's a DNS record created for the domain name.

Cons: Requires the user to use Route53 for their DNS for at least the subdomain that covers this HTTPS endpoint.  Given this is a pattern, I think that's a safe assumption.